### PR TITLE
gha: remove GOPATH and workingdir, update actions/setup-go@v4, actions/checkout@v4

### PR DESCRIPTION
**Pull request by thaJeztah**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,48 +14,39 @@ jobs:
...
 
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
         with:
-          path: src/github.com/containerd/nri
           fetch-depth: 25
 
       - uses: containerd/project-checks@v1
-        with:
-          working-directory: src/github.com/containerd/nri
```
Comment (klihub) [inline .github/workflows/ci.yml:34]: Isn't this the culprit for the DCO check failure ?

**Comment by thaJeztah** [comment]
Looks like the containerd "project" action depends on GOPATH in some way; need to check why (and if we can fix that)

**Comment by klihub** [comment]
> Looks like the containerd "project" action depends on GOPATH in some way; need to check why (and if we can fix that)

Actually it is not the DCO check that failed but the file header check. And looking at the [action](https://github.com/containerd/project-checks/blob/eae3c6e2e734bc70dc79b3d4c3bec73f0f99eeaa/action.yml#L42) it looks like it requires the `working-directory` input to be set. Now I wonder how come it did not fail with an error message complaining about that. Looks like it went berserk instead and scanned all files it could then complained about the lack of the containerd licence header.

**Comment by thaJeztah** [comment]
Yes, I spotted that working-dir "required" property in the project-check repository; so I think that could be related indeed.

Guess I need to look more into that; in the meantime, I reverted the `GOPATH`, `path`, and `working-dir` changes for the project-checks (but kept it for all the other changes). I think that should probably unblock this PR

**Review by klihub — APPROVED** [review]
LGTM.

**Review by dmcgowan — APPROVED** [review]

**Pull request merged**